docs/source/using_websockets.rst

```diff
@@ -4,30 +4,186 @@ Using WebSockets in Dallinger Experiments
+            })
+            .fail(function (rejection) {
+                ...
+            });
+    };
+
+
+When establishing a channel subscription using the `/chat` route, the client can
+include `worker_id` and `participant_id` values which will be included in the
+`"dallinger_control"` channel messages alerting the experiment to WebSocket
+connection, disconnection, subscription, and un-subscription events.
+
+Messages sent over the socket connection can be prefixed with any channel name,
+not just the channel to which the connection is subscribed. Additional
+subscriptions can be established by opening new websocket connections to
+the `/chat` route with different `channel` values.
+
+
+Experiment Channel Setup
+------------------------
+
+Many experiment use cases will only need a "broadcast channel" to which all
+clients subscribe. That subscription can be established when the experiment
+starts (i.e. when `createAgent` returns). This "broadcast channel" would be
+separate from the one set in the `Experiment.channel` attribute, which we will
+call the "experiment control channel".
+
+Clients will receive all messages sent to the "broadcast channel" by either the
+experiment or other clients. The messages will generally contain JSON payloads
+that indicate the messages' purpose. For example, messages may have a `type`
+property to differentiate e.g. "state" messages sent by the experiment server
+from "chat" messages sent by other clients. Additionally, such "chat" messages
+might have `room` or `recipient` properties to allow clients to filter
+out messages not intended for them.
+
+Generally, clients will send messages about their actions to the "experiment
+control channel". Those messages will be processed by the experiment and will
+not be relayed to other clients, because clients are not generally
+subscribed to the "experiment control channel".
+
+The experiment sends messages to all clients over the "broadcast channel", but
+generally does not subscribe to the "broadcast channel". If an experiment needs
+to handle messages sent by clients over the "broadcast channel", then it's
+generally simplest for clients to send such messages both to the "broadcast
+channel" and to the "experiment control channel" (perhaps with an additional
+`broadcast` flag). It is possible to subscribe the experiment to the "broadcast
+channel", but that would also require the experiment to handle/ignore the
+messages that the experiment itself sends over that channel.
+
+
+Multiple Client Channels
+------------------------
+
+If it's important for an experiment to have participant and/or group specific
+channels, e.g. to ensure messages are only ever seen by their targets, or to
+reduce the total number of messages sent to or processed by clients, then
+clients can subscribe to multiple channels.
+
+For example, after launch an experiment could broadcast a `create_chatroom` type
+message with a `chatroom` property set to "room_name" and an array of
```

Comment:
I may not be following the example correctly, but would "room_1" or "room_blu" be a better name here? I understand the idea to be that there are multiple rooms (and multiple room names) and clients should subscribe to messages appropriate to their assigned room.